### PR TITLE
Checkout: Add Shipping Address

### PR DESCRIPTION
**Pull request by nicolas-maalouf-cko**
# Short Summary: 
Add Shipping Address details based on the Checkout.com [API reference](https://api-reference.checkout.com/#operation/requestAPaymentOrPayout)

# Tests:

## Unit Tests:
Finished in 21.159827 seconds.
------------------------------------------------------------------------------------------------------------------------------------------------------------
5492 tests, 77316 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

## Remote Tests:
Finished in 55.268942 seconds.
------------------------------------------------------------------------------------------------------------------------------------------------------------
89 tests, 211 assertions, 1 failures, 3 errors, 0 pendings, 0 omissions, 0 notifications
95.5056% passed

Test that failed is due to Sandbox account not supporting credit (payouts) simulation:
test_successful_credit

### CODE REVIEW

**Comment by heitor-viana-cko** [comment]
Hi @aenand any chance we could get this PR approved please?

**Comment by aenand** [comment]
Hi @heitor-viana-cko could you please provide a bit of context on why you are making the change and how you pull it into your application (i.e. do you pull from `master` or will you be creating a new version)?

**Comment by heitor-viana-cko** [comment]
@aenand this is for Checkout.com's credit card integration in Shopify, we were sending only billing address details but now we need to send shipping address details too!

**Comment by nicolas-maalouf-cko** [comment]
@aenand I just refactored the code a bit and wrote unit and remote tests, let me know if anything else is needed

Thanks a lot for your support on this one

**Comment by aenand** [comment]
> @aenand I just refactored the code a bit and wrote unit and remote tests, let me know if anything else is needed
> 
> Thanks a lot for your support on this one

Thanks for doing this! I'll take a final look at this in a couple days and let you know when we can merge it. Is there a timeline on this request?

**Comment by nicolas-maalouf-cko** [comment]
We have a customer looking to implement fraud rules in our system based on shipping address 

What does the process look like regarding deploying on the Shopify side so we can plan and monitor traffic accordingly? 

Thanks

**Comment by aenand** [comment]
> We have a customer looking to implement fraud rules in our system based on shipping address
> 
> What does the process look like regarding deploying on the Shopify side so we can plan and monitor traffic accordingly?
> 
> Thanks

I'm not sure how things work on the Shopify side. I work for Spreedly and have been a bit more active on this repository lately than they have. You may want to reach out to a contact at Shopify to see how their deployment pipeline works for ActiveMerchant!

**Comment by aenand** [comment]
@heitor-viana-cko I haven't done a release for AM but let me take this back to my team and see if anyone else has!

**Review by aenand — CHANGES_REQUESTED** [review]
~All of the remote tests fail for me on this branch with `ActiveMerchant::ResponseError: Failed with 400 Bad Request`~
I think this is my creds one ⏲️

File: test/remote/gateways/remote_checkout_v2_test.rb [hunk 1]
```diff
@@ -462,6 +467,12 @@ def test_avs_failed_authorize
     assert_equal 'request_invalid: card_number_invalid', response.message
   end
 
+  def test_invalid_shipping_address
+    response = @gateway.authorize(@amount, @credit_card, shipping_address: address.update(country: 'Canada'))
+    assert_failure response
+    assert_equal 'request_invalid: address_country_invalid', response.message
```
Comment (aenand) [inline test/remote/gateways/remote_checkout_v2_test.rb:473]: The gateway responded with this error message when I ran the remote test 🤔 
```suggestion
    assert_equal 'request_invalid: country_address_invalid', response.message
```

**Comment by aenand** [comment]
After getting the right creds, I now am getting the following for remote test results. Gateways can have flaky sandboxes but what is your updated remote test results?
```
91 tests, 219 assertions, 7 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
92.3077% passed
```

File: test/remote/gateways/remote_checkout_v2_test.rb [hunk 2]
```diff
@@ -439,7 +444,7 @@ def test_successful_purchase_with_ip
   end
 
   def test_failed_purchase
-    response = @gateway.purchase(12305, @credit_card, @options)
+    response = @gateway.purchase(100, @credit_card_dnh, @options)
     assert_failure response
     assert_equal 'Declined - Do Not Honour', response.message
```
Comment (aenand) [inline test/remote/gateways/remote_checkout_v2_test.rb:449]: Checkout is responding with a `Invalid Card Number` for this error message 🤔

Comment (nicolas-maalouf-cko) [reply test/remote/gateways/remote_checkout_v2_test.rb:449]: Hi @aenand , I assume you are using legacy API creds that used our old Sandbox which relied on amounts (not card numbers) for simulations, and pointed to our previous gateway iteration which had a few differences in decline codes.

Do you have a processing channel at hand? (starting pc_…)

If not can you please create a new test account using this link please?
https://www.checkout.com/get-test-account

Let me know once the account is created and I’ll make sure it has everything configured for the remote tests to work

Thanks

**Comment by aenand** [comment]
This looks good to me! Please squash your commits into 1 and add a CHANGELOG entry. I will then merge and try to cut a new release on Monday. Since I do not work for Shopify or have access to shipit I cannot verify if the new release triggers something at Shopify.

**Comment by nicolas-maalouf-cko** [comment]
@aenand done, let me know if everything looks good now 

Please keep us posted on the timeframe you'd be creating the release on Monday so we keep an eye on traffic 

Thank you

**Pull request merged**